modules/AlphaWalletFoundation/AlphaWalletFoundation/WalletBalance/WalletBalanceFetcher.swift

```diff
@@ -38,20 +39,35 @@ public class WalletBalanceFetcher: NSObject, WalletBalanceFetcherType {
         self.wallet = wallet
         self.tokensPipeline = tokensPipeline
         self.currencyService = currencyService
+        let provider = ReloadTokensSchedulerProvider(tokensService: tokensService)
+        self.scheduler = Scheduler(provider: provider)
         super.init()
     }
 
     public func start() {
         guard !isRunningTests() else { return }
 
-        timer = Timer.scheduledTimer(withTimeInterval: Self.updateBalanceInterval, repeats: true) { [weak self] _ in
-            self?.refreshBalance(updatePolicy: .all)
-        }
+        scheduler.start()
 
         tokensPipeline.tokenViewModels
             .map { [wallet, currencyService] in WalletBalance(wallet: wallet, tokens: $0, currency: currencyService.currency) }
             .removeDuplicates()
-            .assign(to: \.value, on: walletBalanceSubject)
+            .assign(to: \.value, on: subject)
+            .store(in: &cancelable)
+
+        NotificationCenter.default.applicationState
+            .receive(on: RunLoop.main)
+            .sink { [weak scheduler] state in
+                switch state {
+                case .didEnterBackground:
+                    scheduler?.cancel()
```

Comment:
Does this need to be explicit? Wouldn't it be paused automatically by the OS?

Reply:
no it wouldn't, it still works

Reply:
No I mean. Before the PR, if the app goes into the background, it will pause too, right? What’s the difference?

Reply:
yes, it works in same way with prev code.
added logs to timed call and when app goes to background state
```
xxx.Timer
xxx.Timer
xxx.Timer
xxx.applicationState: didEnterBackground
xxx.applicationState: didEnterBackground
xxx.applicationState: didEnterBackground
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer
xxx.Timer

```